### PR TITLE
Private content contrib flow

### PR DESCRIPTION
**Pull request by guykeller**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
All of the needed changes to get the private repo to it's current status

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/Marketplace/marketplace_services.py [hunk 1]
```diff
@@ -1148,6 +1185,7 @@ def format_metadata(self, user_metadata, pack_content_items, integration_images,
                                                            downloads_count=self.downloads_count,
                                                            is_feed_pack=self._is_feed)
 
+            print_error(f"formatted pack metadata is: {formatted_metadata}")
```
Comment (ronykoz) [inline Tests/Marketplace/marketplace_services.py:1188]: remove

File: Tests/Marketplace/marketplace_services.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import json
 import os
+import stat
```
Comment (ronykoz) [inline Tests/Marketplace/marketplace_services.py:3]: remove all print errors

File: Tests/Marketplace/marketplace_services.py [hunk 3]
```diff
@@ -695,11 +722,15 @@ def zip_pack(self):
                         relative_file_path = os.path.relpath(full_file_path, self._pack_path)
                         pack_zip.write(filename=full_file_path, arcname=relative_file_path)
 
+            if should_encrypt:
+                self.encrypt_pack(zip_pack_path, pack_name, encryption_key, extract_destination_path)
+                zip_pack_path = zip_pack_path.replace("_not_encrypted.zip", ".zip")
```
Comment (ronykoz) [inline Tests/Marketplace/marketplace_services.py:727]: why not having that in the encrypt pack function

File: Tests/Marketplace/zip_packs.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 
 from demisto_sdk.commands.common.tools import print_error, print_success, print_warning, LooseVersion, str2bool
 
-ARTIFACT_NAME = 'content_marketplace_packs.zip'
+ARTIFACT_NAME = 'zipped_packs.zip'
```
Comment (ronykoz) [inline Tests/Marketplace/zip_packs.py:12]: why?

File: Tests/Marketplace/upload_git_snapshot.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    IGNORED_FILES, PACKS_FOLDER, IGNORED_PATHS, Metadata
+from demisto_sdk.commands.common.tools import run_command, print_error, print_warning, print_color, LOG_COLORS, str2bool
+
+
+def upload_git_snapshot(git_snapshot_path, pack_name, branch_name, pack_version, storage_bucket, repo_name, git_org):
+    try:
+        git_snapshot_name = f'org-{git_org}_repo-{repo_name}_branch-{branch_name}_version-{pack_version}.zip'
+        git_snapshot_storage_path = os.path.join('backup', pack_name, pack_version, git_snapshot_name)
+        print_error(f'git_snapshot_storage_path is: {git_snapshot_storage_path}')
+        print_error(f'git_snapshot_path is: {git_snapshot_path}')
```
Comment (ronykoz) [inline Tests/Marketplace/upload_git_snapshot.py:21]: why error?

File: Tests/configure_and_test_integration_instances.py [hunk 6]
```diff
@@ -757,7 +758,14 @@ def set_marketplace_gcp_bucket_for_build(client, prints_manager, branch_name, ci
         'marketplace.initial.sync.delay': '0',
         'content.pack.ignore.missing.warnings.contentpack': 'true'
     }
-    if not is_nightly:
+    if is_private:
+        server_configuration['marketplace.bootstrap.bypass.url'] = 'https://storage.googleapis.com/marketplace-ci-build'
```
Comment (ronykoz) [inline Tests/configure_and_test_integration_instances.py:762]: i think we should combine the actions here with the else, most of them are actually identical, apart from the GW

File: Tests/Marketplace/private_build_prepare_for_testing.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    private_testing_bucket_client = storage_client.bucket(GCPConfig.PRODUCTION_PRIVATE_CI_BUCKET)
+    public_prod_bucket_client = storage_client.bucket(GCPConfig.PRODUCTION_BUCKET)
+
+    extract_packs_artifacts(path_to_artifacts, extract_destination_path)
+    path_to_pack = os.path.join(extract_destination_path, pack_name)
+    premium_pack = Pack(pack_name, path_to_pack)
+    premium_pack.upload_to_storage()
+
+    upload_premium_pack_to_private_testing_bucket(pack_name, private_testing_bucket_client)
+    # index_folder_path, index_blob = download_and_extract_index(public_prod_bucket_client, extract_destination_path)
```
Comment (ronykoz) [inline Tests/Marketplace/private_build_prepare_for_testing.py:88]: remove

File: Tests/Marketplace/prepare_public_index_for_private_testing.py [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    public_storage_bucket = storage_client.bucket(public_bucket_name)
+    private_storage_bucket = storage_client.bucket(private_bucket_name)
+
+    if storage_base_path:
+        GCPConfig.STORAGE_BASE_PATH = storage_base_path
+
+    public_index_folder_path, public_index_blob, _ = download_and_extract_index(public_storage_bucket,
+                                                                                extract_public_index_path)
+    # private_index_folder_path, private_index_blob, _ = download_and_extract_index(private_storage_bucket,
+    #                                                                               extract_private_index_path)
```
Comment (ronykoz) [inline Tests/Marketplace/prepare_public_index_for_private_testing.py:140]: delete

File: Tests/Marketplace/prepare_public_index_for_private_testing.py [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+    # merge_private_index_into_public_index(public_index_folder_path, private_index_folder_path)
+    update_index_with_priced_packs(private_storage_bucket, extract_public_index_path, public_index_folder_path,
+                                   changed_pack, True)
+    change_packs_price_to_zero(public_index_folder_path)
```
Comment (ronykoz) [inline Tests/Marketplace/prepare_public_index_for_private_testing.py:145]: add a comment about that

File: Tests/Marketplace/prepare_public_index_for_private_testing.py [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+def merge_private_index_into_public_index(public_index_folder_path, private_index_folder_path):
+    packs_in_private_index = [pack_dir.name for pack_dir in os.scandir(private_index_folder_path) if pack_dir.is_dir()]
+    for pack_name in packs_in_private_index:
+        path_to_pack_in_private_index = os.path.join(private_index_folder_path, pack_name)
+        path_to_pack_in_public_index = os.path.join(public_index_folder_path, pack_name)
+        shutil.copy(path_to_pack_in_private_index, path_to_pack_in_public_index)
+
+
+def edit_index_json_file(path_to_public_index, build_number):
```
Comment (ronykoz) [inline Tests/Marketplace/prepare_public_index_for_private_testing.py:38]: add docstring, what does this function do

File: Tests/Marketplace/prepare_public_index_for_private_testing.py [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+    path_to_index_json = os.path.join(path_to_public_index, 'index.json')
+    with open(path_to_index_json, 'r') as pack_metadata_file:
+        pack_metadata = json.load(pack_metadata_file)
+
+    if 'packs' in pack_metadata:
+        del pack_metadata['packs']
```
Comment (ronykoz) [inline Tests/Marketplace/prepare_public_index_for_private_testing.py:44]: why are we deleting it

File: Tests/Marketplace/prepare_public_index_for_private_testing.py [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+        current_commit_hash (str): last commit hash of head.
+        index_generation (str): downloaded index generation.
+
+    """
+    with open(os.path.join(public_index_folder_path, "index.json"), "w+") as index_file:
+        index = {
+            'revision': build_number,
+            'modified': datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ')
+        }
+        json.dump(index, index_file, indent=4)
```
Comment (ronykoz) [inline Tests/Marketplace/prepare_public_index_for_private_testing.py:70]: where are the packs here

File: Tests/Marketplace/upload_packs.py [hunk 13]
```diff
@@ -752,141 +945,51 @@ def main():
...
     if private_bucket_name:  # Add private packs to the index
-        private_storage_bucket = storage_client.bucket(private_bucket_name)
-        private_packs = update_index_with_priced_packs(private_storage_bucket, extract_destination_path,
-                                                       index_folder_path)
+        private_packs, private_index_path, private_index_blob = update_index_with_priced_packs(private_storage_bucket,
+                                                                                               extract_destination_path,
+                                                                                               index_folder_path,
+                                                                                               pack_names,
+                                                                                               is_private_build)
+        print_error(f"private packs are: {private_packs}")
```
Comment (ronykoz) [inline Tests/Marketplace/upload_packs.py:958]: remove

File: Tests/Marketplace/upload_packs.py [hunk 14]
```diff
@@ -411,17 +454,20 @@ def update_index_with_priced_packs(private_storage_bucket, extract_destination_p
     private_packs = []
 
     try:
-        private_index_path, _, _ = download_and_extract_index(private_storage_bucket,
-                                                              os.path.join(extract_destination_path, 'private'))
-        private_packs = get_private_packs(private_index_path)
+        private_index_path, private_index_blob, _ = download_and_extract_index(private_storage_bucket,
+                                                                               os.path.join(extract_destination_path,
+                                                                                            'private'))
+        print_error(f"ls: {subprocess.check_output('ls')}")
```
Comment (ronykoz) [inline Tests/Marketplace/upload_packs.py:460]: remove

File: Tests/Marketplace/upload_packs.py [hunk 15]
```diff
@@ -411,17 +454,20 @@ def update_index_with_priced_packs(private_storage_bucket, extract_destination_p
...
+        private_packs = get_private_packs(private_index_path, pack_names, is_private_build, extract_destination_path)
         add_private_packs_to_index(index_folder_path, private_index_path)
         print("Finished updating index with priced packs")
     except Exception as e:
         print_error(f'Could not add private packs to the index: {str(e)}')
     finally:
-        if private_index_path:
-            shutil.rmtree(os.path.dirname(private_index_path), ignore_errors=True)
-        return private_packs
+        # if private_index_path:
```
Comment (ronykoz) [inline Tests/Marketplace/upload_packs.py:468]: remove

File: Tests/Marketplace/upload_packs.py [hunk 16]
```diff
@@ -752,141 +945,51 @@ def main():
...
+                                           packs_dependencies_mapping,
+                                           private_storage_bucket=private_storage_bucket, content_repo=content_repo,
+                                           current_commit_hash=current_commit_hash,
+                                           remote_previous_commit_hash=remote_previous_commit_hash)
     # upload core packs json to bucket
-    upload_core_packs_config(storage_bucket, build_number, index_folder_path)
 
+    if should_upload_core_packs(storage_bucket_name):
+        upload_core_packs_config(default_storage_bucket, build_number, index_folder_path)
     # finished iteration over content packs
```
Comment (ronykoz) [inline Tests/Marketplace/upload_packs.py:980]: seems irrelevant

File: Tests/scripts/run_private_build_tests.sh [hunk 17]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
```
Comment (ronykoz) [inline Tests/scripts/run_private_build_tests.sh:1]: how is it different from the one we have in content ?

File: Tests/scripts/prepare_private_content_packs_for_testing.sh [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+CONTENT_PACKS_TO_INSTALL_FILE="./Tests/content_packs_to_install.txt"
+if [ ! -f $CONTENT_PACKS_TO_INSTALL_FILE ]; then
+  echo "Could not find file $CONTENT_PACKS_TO_INSTALL_FILE."
+else
+  CONTENT_PACKS_TO_INSTALL=$(paste -sd, $CONTENT_PACKS_TO_INSTALL_FILE)
+  if [[ -z "$CONTENT_PACKS_TO_INSTALL" ]]; then
+    echo "Did not get content packs to update in the bucket."
+  else
+    echo "Updating the following content packs: $CONTENT_PACKS_TO_INSTALL ..."
+    # python3 ./Tests/Marketplace/upload_packs.py -b $GCS_TESTING_BUCKET -pb $GCS_PRIVATE_TESTING_BUCKET -a $PACK_ARTIFACTS -d $CIRCLE_ARTIFACTS/packs_dependencies.json -e $EXTRACT_FOLDER -s $KF -n $GITHUB_RUN_NUMBER -p $NEW_PACK_NAME -sb $PUBLIC_TARGET_PATH -k $PACK_SIGN_KEY -rt false --id_set_path $ID_SET -pr True -ek $PACK_ENCRYPTION_KEY -enc True -o
```
Comment (ronykoz) [inline Tests/scripts/prepare_private_content_packs_for_testing.sh:64]: remove

File: Tests/scripts/prepare_private_content_packs_for_testing.sh [hunk 19]
```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
```
Comment (ronykoz) [inline Tests/scripts/prepare_private_content_packs_for_testing.sh:1]: not sure why we need another file like this, why can't we configure it all the same way?
the build shouldn't install private content anyway so it doesn't matter

**Review by ronykoz — CHANGES_REQUESTED** [review]
Why can't we load the metadata one time and then write it back at the end of messing with it instead of opening the metadata all the time

File: Tests/Marketplace/marketplace_services.py [hunk 20]
```diff
@@ -35,6 +36,8 @@ class GCPConfig(object):
     USE_GCS_RELATIVE_PATH = True  # whether to use relative path in uploaded to gcs images
     GCS_PUBLIC_URL = "https://storage.googleapis.com"  # disable-secrets-detection
     PRODUCTION_BUCKET = "marketplace-dist"
+    PRODUCTION_PRIVATE_BUCKET = "marketplace-dist-private"
+    PRODUCTION_PRIVATE_CI_BUCKET = "marketplace-ci-build-private"
```
Comment (yaakovi) [inline Tests/Marketplace/marketplace_services.py:40]: so... it's not production?

File: Tests/scripts/run_private_build_tests.sh [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+
+echo "start content tests"
+
+SECRET_CONF_PATH=$(cat secret_conf_path)
+CONF_PATH="./Tests/conf.json"
+DEMISTO_API_KEY=$(cat $SECRET_CONF_PATH | jq '.temp_apikey')
+
+temp="${DEMISTO_API_KEY%\"}"
+temp="${temp#\"}"
+DEMISTO_API_KEY=$temp
```
Comment (yaakovi) [inline Tests/scripts/run_private_build_tests.sh:11]:
```suggestion
DEMISTO_API_KEY=$(cat $SECRET_CONF_PATH | jq -r '.temp_apikey')
```

File: Tests/scripts/run_private_build_tests.sh [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+
+[ -n "${NIGHTLY}" ] && IS_NIGHTLY=true || IS_NIGHTLY=false
+[ -n "${MEM_CHECK}" ] && MEM_CHECK=true || MEM_CHECK=false
+
+code_1=0
+code_2=0
+
+echo "starting configure_and_test_integration_instances"
+# PREVIOUS_JOB_NUMBER=`cat create_instances_build_num.txt` TODO : remove
```
Comment (yaakovi) [inline Tests/scripts/run_private_build_tests.sh:20]: TODO?

File: Tests/scripts/run_private_build_tests.sh [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+
+python3 ./Tests/configure_and_test_integration_instances.py -u "$USERNAME" -p "$PASSWORD" -c "$CONF_PATH" -s "$SECRET_CONF_PATH" -g "$GIT_SHA1" --ami_env "$1" -n $IS_NIGHTLY --branch "$BRANCH_NAME" --build-number "$GITHUB_RUN_NUMBER" -pr true
+code_1=$?
+
+# echo 'export GOOGLE_APPLICATION_CREDENTIALS="creds.json"' >> $BASH_ENV TODO - commented
```
Comment (yaakovi) [inline Tests/scripts/run_private_build_tests.sh:25]: TODO?

File: Tests/scripts/run_private_build_tests.sh [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+echo ::set-env name=GOOGLE_APPLICATION_CREDENTIALS::"creds.json"
+# source $BASH_ENV TODO - commented
```
Comment (yaakovi) [inline Tests/scripts/run_private_build_tests.sh:27]: TODO?

File: Tests/scripts/run_private_build_tests.sh [hunk 25]
```diff
@@ -0,0 +1,47 @@
...
+  if [ -n "${NON_AMI_RUN}" ]; then
+    # non AMI
+    python3 ./Tests/test_content.py -k "$DEMISTO_API_KEY" -c "$CONF_PATH" -e "$SECRET_CONF_PATH" -n $IS_NIGHTLY -t "$SLACK_TOKEN" -a "$CIRCLECI_TOKEN" -b "$GITHUB_RUN_NUMBER" -g "$BRANCH_NAME" -m "$MEM_CHECK" -d "$1"
+  else
+    # AMI
+    python3 ./Tests/test_content.py -k "$DEMISTO_API_KEY" -c "$CONF_PATH" -e "$SECRET_CONF_PATH" -n $IS_NIGHTLY -t "$SLACK_TOKEN" -a "$CIRCLECI_TOKEN" -b "$GITHUB_RUN_NUMBER" -g "$BRANCH_NAME" -m "$MEM_CHECK" --isAMI true -d "$1"
+  fi
+fi
+
+code_2=$?
```
Comment (yaakovi) [inline Tests/scripts/run_private_build_tests.sh:43]: was it removed on purpose?
```suggestion
code_2=$?

if [ $code_1 -eq 0 ] && [ $code_2 -eq 0 ] ; then
  role="$(echo -e "${INSTANCE_ROLE}" | tr -d '[:space:]')"
  filepath="./Tests/is_build_passed_${role}.txt"
  touch "$filepath"
fi
```

File: Tests/scripts/run_private_build_tests.sh (hunk 17 above)
Comment (yaakovi) [reply Tests/scripts/run_private_build_tests.sh:1]: you can merge this with the current script.
extract the relevant changes as script arguments (CircleCI vs. GitHub actions)

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 26]
```diff
@@ -1259,7 +1259,10 @@ def create_test_file(is_nightly, skip_save=False):
         branch_name = branch_name_reg.group(1)
 
         logging.info("Getting changed files from the branch: {0}".format(branch_name))
-        if branch_name != 'master':
+        if file_string:
+            print(f"entered file string if statement. file string is: {file_string}")
+            files_string = file_string
```
Comment (yaakovi) [inline Tests/scripts/collect_tests_and_content_packs.py:1264]: very confusing naming choices

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 27]
```diff
@@ -1303,10 +1306,11 @@ def create_test_file(is_nightly, skip_save=False):
     parser.add_argument('-n', '--nightly', type=tools.str2bool, help='Is nightly or not')
     parser.add_argument('-s', '--skip-save', type=tools.str2bool,
                         help='Skipping saving the test filter file (good for simply doing validation)')
+    parser.add_argument('-f', '--file-string', type=str, help='Is nightly or not')
```
Comment (yaakovi) [inline Tests/scripts/collect_tests_and_content_packs.py:1309]: fix help

File: Tests/Marketplace/zip_packs.py [hunk 28]
```diff
@@ -56,6 +57,8 @@ def zip_packs(packs, destination_path):
     with ZipFile(os.path.join(destination_path, ARTIFACT_NAME), mode='w') as zf:
         for zip_pack in packs:
             for name, path in zip_pack.items():
+                if name not in packs_to_include:
```
Comment (yaakovi) [inline Tests/Marketplace/zip_packs.py:60]: `packs_to_include` might be empty as it is not a required argument.
add a check like [here](https://github.com/demisto/content/pull/8741/files?file-filters%5B%5D=#diff-07f5616b9b8d97b7b823933458483095R131)

File: Tests/scripts/prepare_private_content_packs_for_testing.sh (hunk 19 above)
Comment (guykeller) [reply Tests/scripts/prepare_private_content_packs_for_testing.sh:1]: There are 2 ways to do this - either make all the env variables parameters, or save every single env variable in github actions with the same name as the equivalent circle env var.
Neither of those feels right to me - do you have an idea for a better way?
Also applies to `run_private_build_tests`.
@yaakovi would love your opinion here as well

Comment (yaakovi) [reply Tests/scripts/prepare_private_content_packs_for_testing.sh:1]: as I commented here:
https://github.com/demisto/content/pull/8741#discussion_r480474726
no need to extract all variables, only a few (2-4).

**Review by ronykoz — COMMENTED** [review]
Not sure why you reverted the changes in zip_packs

File: Tests/scripts/prepare_private_content_packs_for_testing.sh (hunk 19 above)
Comment (guykeller) [reply Tests/scripts/prepare_private_content_packs_for_testing.sh:1]: After looking into the logic of this script, this isn't just a matter of variables.
The whole logic of working with the buckets here is different.
Changing it would just mean using a lot of booleans and if statements, and every change to the public one might break the private build.
Your call @ronykoz @yaakovi

File: Tests/configure_and_test_integration_instances.py [hunk 29]
```diff
@@ -757,7 +758,14 @@ def set_marketplace_gcp_bucket_for_build(client, prints_manager, branch_name, ci
...
+        server_configuration['marketplace.gcp.path'] = 'content/builds/{}/{}/content/packs'.format(branch_name,
```
Comment (IkaDemisto) [inline Tests/configure_and_test_integration_instances.py:763]: Same question about the location of `dummy` index. I remember that we discussed on updating same index for private packs testing on static location. I probably miss something

**Review by IkaDemisto — COMMENTED** [review]
Looks good 💰 
Have some comments

File: Tests/Marketplace/marketplace_services.py [hunk 30]
```diff
@@ -677,14 +685,26 @@ def sign_pack(self, signature_string=None):
...
+    def encrypt_pack(self, zip_pack_path, pack_name, encryption_key, extract_destination_path):
+
+        shutil.copy('./encryptor', os.path.join(extract_destination_path, 'encryptor'))
+        os.chmod(os.path.join(extract_destination_path, 'encryptor'), stat.S_IXOTH)
+        current_working_dir = os.getcwd()
+        os.chdir(extract_destination_path)
+        output_file = zip_pack_path.replace("_not_encrypted.zip", ".zip")
+        subprocess.call('chmod +x ./encryptor', shell=True)
+        full_command = f'./encryptor ./{pack_name}_not_encrypted.zip {output_file} "{encryption_key}"'
+        subprocess.call(full_command, shell=True)
```
[the remark this answers is not in the document]
Comment (anara123) [reply Tests/Marketplace/marketplace_services.py:699]: Make sure the packs are encrypted. I downloaded few packs zip created by private build and they were not encrypted

File: Tests/Marketplace/upload_packs_private.py [hunk 31]
```diff
@@ -290,6 +290,7 @@ def upload_index_to_storage(index_folder_path, extract_destination_path, index_b
         index_blob.cache_control = "no-cache,max-age=0"  # disabling caching for index blob
 
         if is_private or current_index_generation == index_generation:
+            print_error("uploading index")
```
Comment (ronykoz) [inline Tests/Marketplace/upload_packs_private.py:293]: remove this line

File: Tests/scripts/download_pack_artifacts.py [hunk 32]
```diff
@@ -0,0 +1,2 @@
+with open('pack_list.txt', 'r') as f:
```
Comment (ronykoz) [inline Tests/scripts/download_pack_artifacts.py:1]: where is this being used

File: Tests/Marketplace/search_and_install_packs.py [hunk 33]
```diff
@@ -178,42 +179,48 @@ def install_packs(client, host, prints_manager, thread_index, packs_to_install,
...
+    if private_install:
+        local_packs = glob.glob("/home/runner/work/content-private/content-private/content/artifacts/packs/*.zip")
+        for local_pack in local_packs:
+            if any(pack_to_install['id'] in local_pack for pack_to_install in packs_to_install):
+                upload_zipped_packs(client=client, host=host, prints_manager=prints_manager, thread_index=thread_index, pack_path=local_pack)
+    else:
+        request_data = {
+            'packs': packs_to_install,
+            'ignoreWarnings': True
+        }
```
Comment (ronykoz) [inline Tests/Marketplace/search_and_install_packs.py:191]: why are we installing all the packs?

File: Tests/Marketplace/upload_packs_private.py [hunk 34]
```diff
@@ -909,6 +907,14 @@ def option_handler():
...
 
 def main():
+    packs_dir = '/home/runner/work/content-private/content-private/content/artifacts/packs'
+    zip_path = '/home/runner/work/content-private/content-private/content/temp-dir'
+    if not os.path.exists(packs_dir):
+        print("Packs dir not found. Creating.")
+        os.mkdir(packs_dir)
+    if not os.path.exists(zip_path):
+        print("Temp dir not found. Creating.")
+        os.mkdir(zip_path)
```
Comment (ronykoz) [inline Tests/Marketplace/upload_packs_private.py:917]: split to another func

File: Tests/configure_and_test_integration_instances.py [hunk 35]
```diff
@@ -901,18 +902,11 @@ def configure_servers_and_restart(build, prints_manager):
...
-            set_docker_hardening_for_build(client, prints_manager)
-            if LooseVersion(build.server_numeric_version) >= LooseVersion('6.0.0'):
-                set_marketplace_gcp_bucket_for_build(client, prints_manager, build.branch_name,
-                                                     build.ci_build_number, build.is_nightly, build.is_private)
-
-            if Build.run_environment == Running.WITH_LOCAL_SERVER:
-                input('restart your server and then press enter.')
-            else:
-                restart_server(server)
-        if Build.run_environment != Running.WITH_LOCAL_SERVER:
```
Comment (ronykoz) [inline Tests/configure_and_test_integration_instances.py]: why is this removed

File: Tests/Marketplace/search_and_install_packs.py (hunk 33 above)
Comment (amshamah419) [reply Tests/Marketplace/search_and_install_packs.py:191]: It's just installing the packs which are being tested. There currently is an issue where the premium packs have to be installed manually due to limitations in setting the marketplace URL for premium

Comment (ronykoz) [reply Tests/Marketplace/search_and_install_packs.py]: Yeah but this line gets all of the content packs, no?
        local_packs = glob.glob("/home/runner/work/content-private/content-private/content/artifacts/packs/*.zip")

Comment (amshamah419) [reply Tests/Marketplace/search_and_install_packs.py]: That line grabs all of the packs which we have locally in the artifacts that we have available to upload. Then we loop through them and if they are in the list of packs_to_install then they get uploaded. If not it just continues the loop.

File: Tests/Marketplace/upload_packs_private.py (hunk 31 above)
Comment (ronykoz) [reply Tests/Marketplace/upload_packs_private.py:292]: remove this line please

File: Tests/scripts/run_private_build_tests.sh (hunk 22 above)
Comment (ronykoz) [reply Tests/scripts/run_private_build_tests.sh:20]: remove this line

File: Tests/scripts/run_private_build_tests.sh (hunk 23 above)
Comment (ronykoz) [reply Tests/scripts/run_private_build_tests.sh:25]: here too

File: Tests/scripts/run_private_build_tests.sh (hunk 24 above)
Comment (ronykoz) [reply Tests/scripts/run_private_build_tests.sh:27]: here too

File: Tests/scripts/run_private_build_tests.sh (hunk 25 above)
Comment (ronykoz) [reply Tests/scripts/run_private_build_tests.sh:43]: @guykeller ?

**Review by ronykoz — COMMENTED** [review]
last couple of questions and we are good to go

**Review by ronykoz — APPROVED** [review]

File: Tests/Marketplace/zip_packs.py [hunk 36]
```diff
@@ -133,9 +133,6 @@ def download_packs_from_gcp(storage_bucket, gcp_path, destination_path, circle_b
             else:
                 pack_prefix = os.path.join(gcp_path, branch_name, circle_build, pack.name)
 
-            if not branch_name or not circle_build:
```
[the remark this answers is not in the document]
Comment (yaakovi) [reply Tests/Marketplace/zip_packs.py:136]: looks like a bug 😜
